static/js/pages/SitePage.tsx

```diff
@@ -23,36 +24,35 @@ export default function SitePage(props: SitePageProps): JSX.Element | null {
 
   const website = useWebsite()
 
-  if (isLoading) {
-    return <div className="site-page std-page-body container">Loading...</div>
-  }
```

Comment:
This early return was causing unnecessary component mount/unmounts lower in the component tree. For example:
1. Site is loaded, `<SiteContentListing />` renders
2. click publish, new site api request is sent

    **Before:** this `<div className="site-page std-page-body container">Loading...</div>` would replace the content listing, then a new content listing would be mounted after the API request is finished.

    **Now:** while the API request is in flight, the old content does not unmount (but a loading spinner shows atop it). When the API request finishes, the child component is updated (no unmount/remount, which was causing dirty form data to be lost).